apiserver/test/e2e/config_server_e2e_test.go

```diff
@@ -0,0 +1,299 @@
+			}
+		})
+	}
+}
+
+// TestDeleteTemplate sequentially iterates over the delete compute template endpoint
+// to validate various input scenarios
+func TestDeleteTemplate(t *testing.T) {
+	tCtx, err := NewEnd2EndTestingContext(t)
+	require.NoError(t, err, "No error expected when creating testing context")
+
+	tCtx.CreateComputeTemplate(t)
+
+	tests := []GenericEnd2EndTest[*api.DeleteComputeTemplateRequest]{
+		{
+			Name: "Delete existing template",
+			Input: &api.DeleteComputeTemplateRequest{
+				Name:      tCtx.GetComputeTemplateName(),
+				Namespace: tCtx.GetNamespaceName(),
+			},
+			ExpectedError: nil,
+		},
+		{
+			Name: "Delete non existing template",
+			Input: &api.DeleteComputeTemplateRequest{
+				Name:      "another-template",
+				Namespace: tCtx.GetNamespaceName(),
+			},
+			ExpectedError: &kuberayHTTP.KuberayAPIServerClientError{
+				HTTPStatusCode: http.StatusNotFound,
+			},
+		},
+		{
+			Name: "Delete a template with no name",
+			Input: &api.DeleteComputeTemplateRequest{
+				Namespace: tCtx.GetNamespaceName(),
+			},
+			ExpectedError: &kuberayHTTP.KuberayAPIServerClientError{
+				HTTPStatusCode: http.StatusBadRequest,
+			},
+		},
+		{
+			Name: "Delete a template with no namespace",
+			Input: &api.DeleteComputeTemplateRequest{
+				Name: "some-name",
+			},
+			ExpectedError: &kuberayHTTP.KuberayAPIServerClientError{
+				HTTPStatusCode: http.StatusNotFound,
+			},
+		},
+	}
+	// Execute tests sequentially
+	for _, tc := range tests {
+		tc := tc // capture range variable
+		t.Run(tc.Name, func(t *testing.T) {
+			actualRpcStatus, err := tCtx.GetRayApiServerClient().DeleteComputeTemplate(tc.Input)
+			if tc.ExpectedError != nil {
+				require.EqualError(t, err, tc.ExpectedError.Error(), "Matching error expected")
+				require.NotNil(t, actualRpcStatus, "A not nill RPC status is required")
+			} else {
+				require.NoError(t, err, "No error expected")
+				require.Nil(t, actualRpcStatus, "No RPC status expected")
+			}
+		})
+	}
+}
+
+// TestGetAllTheTemplates tests gets all compute templates endpoint
+// to validate various input scenarios
+func TestGetAllComputeTemplates(t *testing.T) {
+	tCtx, err := NewEnd2EndTestingContext(t)
+	require.NoError(t, err, "No error expected when creating testing context")
+
+	tCtx.CreateComputeTemplate(t)
+	t.Cleanup(func() {
+		tCtx.DeleteComputeTemplate(t)
+	})
+
+	response, actualRpcStatus, err := tCtx.GetRayApiServerClient().GetAllComputeTemplates()
+	require.NoError(t, err, "No error expected")
+	require.Nil(t, actualRpcStatus, "No RPC status expected")
+	require.NotNil(t, response, "A response is expected")
+	require.NotEmpty(t, response.ComputeTemplates, "A list of compute templates is required")
+	require.Equal(t, tCtx.GetComputeTemplateName(), response.ComputeTemplates[0].Name)
+	require.Equal(t, tCtx.GetNamespaceName(), response.ComputeTemplates[0].Namespace)
+}
```

Comment:
I failed this test when I ran all test from end to end however moving this test to the beginning  resolved the problem for me.

Here is is the partial apiserver log:
```
...
I1031 20:00:27.480132       1 interceptor.go:14] /proto.ComputeTemplateService/CreateComputeTemplate handler starting
I1031 20:00:27.556904       1 interceptor.go:19] /proto.ComputeTemplateService/CreateComputeTemplate handler finished
I1031 20:00:27.558489       1 interceptor.go:14] /proto.ComputeTemplateService/ListAllComputeTemplates handler starting
W1031 20:00:28.559590       1 interceptor.go:17] List all compute templates from all namespaces failed.: List compute templates failed in kube-public: context canceled
I1031 20:00:28.559625       1 interceptor.go:19] /proto.ComputeTemplateService/ListAllComputeTemplates handler finished
I1031 20:00:28.561052       1 interceptor.go:14] /proto.ComputeTemplateService/DeleteComputeTemplate handler starting
I1031 20:00:28.957753       1 interceptor.go:19] /proto.ComputeTemplateService/DeleteComputeTemplate handler finished
```

go test message:
```
go test ./test/e2e/...  -timeout 60m  -race -count=1 -parallel 4


--- FAIL: TestGetAllComputeTemplates (1.80s)
    config_server_e2e_test.go:206:
        	Error Trace:	/root/kuberay/apiserver/test/e2e/config_server_e2e_test.go:206
        	Error:      	Received unexpected error:
        	            	failed to execute http request for url 'http://localhost:31888/apis/v1/compute_templates': Get "http://localhost:31888/apis/v1/compute_templates": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
        	Test:       	TestGetAllComputeTemplates
        	Messages:   	No error expected
```

Reply:
Moving the test at the beginning will just mask the underlying problem uncovered by the test. I don't think we should fix the issue in this PR. I have created: #1591 to track this separately. I think we should merge this PR.